app.js

```diff
@@ -13,6 +14,14 @@ var app = express();
 app.set('views', path.join(__dirname, 'views'));
 app.set('view engine', 'pug');
 
+// save user session. not working!!!!
+app.use(session({
```

Comment:
You may not be seeing a cookie since `localhost` (127.0.0.1) is insecure (http). the `secure: true` option on the cookie looks like it's trying to set a cookie on secure (https). I just did a quick skim of the docs, but that's my first inclination of what is going wrong.

You can try this to set secure cookies in production but allow for local development:
```js
var sess = {
  secret: 'keyboard cat',
  cookie: {}
}

if (app.get('env') === 'production') {
  app.set('trust proxy', 1) // trust first proxy
  sess.cookie.secure = true // serve secure cookies
}

app.use(session(sess))
```

Reply:
This is working. I'll spend more time reading the documentation. I appreciate the detailed review.